grpc/dial.go

```diff
@@ -7,9 +7,15 @@ import (
 
 	"github.com/edaniels/golog"
 	"go.viam.com/utils/rpc"
+
+	"go.viam.com/rdk/utils/contextutils"
 )
 
-// Dial dials a gRPC server.
+// defaultDialTimeout is the default timeout for dialing a robot.
+var defaultDialTimeout = 20 * time.Second
+
+// Dial dials a gRPC server. `ctx` can be used to set a timeout/deadline for Dial.
+// However, the max timeout is 20 seconds.
```

Comment:
```suggestion
However, the signaling server may have other timeouts which may prevent the full timeout from being respected.
```
on second thought, maybe we shouldn't say that we have a max of 20s - we technically don't know that for sure (we may change logic down the line, for example).

Reply:
@andf-viam I think the "max" is 20 seconds. I'm not sure why exactly . Seems like there is out-of-scope code in the signaling server that prevents it from being more